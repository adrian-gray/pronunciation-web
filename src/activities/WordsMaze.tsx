import React, { useState } from "react";
import { IWordsMaze } from "./../../@types/PronounceWeb";
import Script from "react-load-script";

import MemberGate from "./../components/MemberGate";
import SplitHilite from "./../components/SplitHilite";

import WordsMaze from "./WordsMaze/index";

const NATIVE_WIDTH = 750;
const NATIVE_HEIGHT = 750;

export default (props: IWordsMaze) => {
  let screenWidth = NATIVE_WIDTH;
  let screenHeight = NATIVE_HEIGHT;

  const handleScriptLoad = () => {
    const wordsMaze = Object.create(WordsMaze);
    wordsMaze.init({
      words: props.words,
      correct: props.correct,
      auth: props.isUserAuth
    });
  };

  const _handleResize = () => {
    const padding = 30;
    const w = window.innerWidth - padding;
    const scale = w / NATIVE_WIDTH;
    screenWidth = Math.round(scale * NATIVE_WIDTH);
    screenHeight = Math.round(scale * NATIVE_HEIGHT);
  };
  window.addEventListener("resize", _handleResize);

  const width = `${screenWidth}px`;
  const height = `${screenHeight}px`;
  const style = { width, height };
  const title = `Words Maze ${props.tag} sound – ${props.ipa}`;
  const description = `Discover your way through the maze. Find the words that share the ${props.tag} sound and travel along the path of the maze from one word to the next until you discover the way out. Master each level and advance to the next – more complex words, more twists and turns. Maze is an interactive pronunciation game that helps you become familiar with common English words so you can speak with confidence.`;
  const instructions = `From the yellow square pick an adjacent square with a ${props.ipa} sound. Keep going until you reach the blue square.`;
  const display = (
    <div>
      <Script
        onLoad={handleScriptLoad}
        onError={() => console.error("scriprt error")}
        url="https://cdnjs.cloudflare.com/ajax/libs/phaser/2.6.2/phaser.min.js"
      />
      <div id="words-maze" style={style} />
    </div>
  );

  return (
    <div className="head-space">
      <h3>
        <SplitHilite str={title} />
      </h3>
      <p>{description}</p>
      <p>{instructions}</p>
      <MemberGate content={display} isUserAuth={props.isUserAuth} />
    </div>
  );
};
