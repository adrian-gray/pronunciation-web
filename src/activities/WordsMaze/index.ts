/// <reference path="./../../../@types/phaser/phaser.d.ts" />
import Boot from "./states/boot";
import Game from "./states/game";

import { IGame } from "./../../../@types/PronounceWeb";

let game: IGame;

interface IWordsMaze {
  words: string[];
  correct: string[];
  auth: number;
}

const WordsMaze = {
  init(params: IWordsMaze) {
    game = new Phaser.Game(750, 750, Phaser.AUTO, "words-maze");
    game.data = params;
    game.isAuth = !!params.auth;
    game.state.add("boot", Boot);
    game.state.add("game", Game);
    game.state.start("boot");
    return game;
  },

  destroy() {
    if (game) {
      game.destroy();
    }
    game = null;
  }
};

export default WordsMaze;
