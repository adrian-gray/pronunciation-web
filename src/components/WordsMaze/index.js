/* global Phaser */

import Boot from "./states/boot";
import Game from "./states/game";

let game;

const WordsMaze = {
  init(params) {
    game = new Phaser.Game(750, 750, Phaser.AUTO, "words-maze");
    game.data = params;
    game.isAuth = params.auth;
    game.state.add("boot", Boot);
    game.state.add("game", Game);
    game.state.start("boot");
    return game;
  },

  destroy() {
    if (game) {
      game.destroy();
    }
    game = null;
  }
};

export default WordsMaze;
