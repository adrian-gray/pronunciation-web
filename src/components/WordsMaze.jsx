import React, { Component } from "react";
import Script from "react-load-script";

import MemberGate from "./MemberGate";
import SplitHilite from "./SplitHilite";
import WordsMaze from "./WordsMaze/index";

import Typography from "@material-ui/core/Typography";
import withStyles from "@material-ui/styles/withStyles";

const styles = theme => ({
  headspace: theme.headspace
});

const NATIVE_WIDTH = 750;
const NATIVE_HEIGHT = 750;

class Maze extends Component {
  constructor(props) {
    super(props);

    const other = { ...props };

    this.state = {
      other,
      words: props.words,
      correct: props.correct
    };

    this.handleScriptError = this.handleScriptError.bind(this);
    this.handleScriptLoad = this.handleScriptLoad.bind(this);
    this._handleResize = this.handleResize.bind(this);
  }

  componentDidMount() {
    this.handleResize();
    this.windowListener = window.addEventListener("resize", this._handleResize);
  }

  componentWillUnmount() {
    window.removeEventListener("resize", this._handleResize);

    if (this && this.state && this.state.game) {
      this.state.game.destroy();
    } else {
      console.log("Unsuccessfully tried to destroy Words Maze", this.state);
    }
  }

  handleScriptError() {
    this.forceUpdate();
  }

  handleScriptLoad() {
    const wordsMaze = Object.create(WordsMaze);
    this.setState({
      game: wordsMaze
    });

    if (this && this.state) {
      wordsMaze.init({
        words: this.state.words,
        correct: this.state.correct,
        auth: this.props.userAuth
      });
    } else {
      console.log("Words Maze state no longer exists, move on without creating game");
    }
  }

  handleResize() {
    const padding = "30px";
    const w = window.innerWidth - padding;
    const scale = w / NATIVE_WIDTH;
    const width = Math.round(scale * NATIVE_WIDTH);
    const height = Math.round(scale * NATIVE_HEIGHT);
    this.setState({ width, height });
  }

  render() {
    const width = `${this.state.width}px`;
    const height = `${this.state.height}px`;
    const style = { width, height };
    const title = `Words Maze ${this.props.tag} sound – ${this.props.ipa}`;
    const description = `Discover your way through the maze. Find the words that share the ${
      this.props.tag
    } sound and travel along the path of the maze from one word to the next until you discover the way out. Master each level and advance to the next – more complex words, more twists and turns. Maze is an interactive pronunciation game that helps you become familiar with common English words so you can speak with confidence.`;
    const instructions = `From the yellow square pick an adjacent square with a ${
      this.props.ipa
    } sound. Keep going until you reach the blue square.`;
    const display = (
      <div>
        <Script
          onLoad={this.handleScriptLoad}
          onError={this.handleScriptError}
          url="https://cdnjs.cloudflare.com/ajax/libs/phaser/2.6.2/phaser.min.js"
        />
        <div id="words-maze" style={style} />
      </div>
    );

    return (
      <div className={this.props.classes.headspace}>
        <Typography variant="title" gutterBottom>
          <SplitHilite str={title} />
        </Typography>
        <Typography variant="body2" gutterBottom>
          {description}
        </Typography>
        <Typography variant="body2" gutterBottom>
          {instructions}
        </Typography>
        <MemberGate content={display} userAuth={this.props.userAuth} {...this.other} />
      </div>
    );
  }
}

export default withStyles(styles)(Maze);
