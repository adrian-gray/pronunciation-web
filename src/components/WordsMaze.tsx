import React, { useState } from "react";
import Script from "react-load-script";

import Typography from "@material-ui/core/Typography";
import { makeStyles } from "@material-ui/core/styles";
import { withTheme } from "@material-ui/styles";

import MemberGate from "./MemberGate";
import SplitHilite from "./SplitHilite";
import WordsMaze from "./WordsMaze/index";

const useStyles = makeStyles(theme => ({
  headspace: theme.headspace
}));

const NATIVE_WIDTH = 750;
const NATIVE_HEIGHT = 750;

function Maze(props) {
  const classes = useStyles(props);

  let screenWidth = NATIVE_WIDTH;
  let screenHeight = NATIVE_HEIGHT;

  const handleScriptLoad = () => {
    const wordsMaze = Object.create(WordsMaze);
    wordsMaze.init({
      words: props.words,
      correct: props.correct,
      auth: props.userAuth
    });
  };

  const _handleResize = () => {
    const padding = 30;
    const w = window.innerWidth - padding;
    const scale = w / NATIVE_WIDTH;
    screenWidth = Math.round(scale * NATIVE_WIDTH);
    screenHeight = Math.round(scale * NATIVE_HEIGHT);
  };
  window.addEventListener("resize", _handleResize);

  const width = `${screenWidth}px`;
  const height = `${screenHeight}px`;
  const style = { width, height };
  const title = `Words Maze ${props.tag} sound – ${props.ipa}`;
  const description = `Discover your way through the maze. Find the words that share the ${props.tag} sound and travel along the path of the maze from one word to the next until you discover the way out. Master each level and advance to the next – more complex words, more twists and turns. Maze is an interactive pronunciation game that helps you become familiar with common English words so you can speak with confidence.`;
  const instructions = `From the yellow square pick an adjacent square with a ${props.ipa} sound. Keep going until you reach the blue square.`;
  const display = (
    <div>
      <Script
        onLoad={handleScriptLoad}
        onError={() => console.error("scriprt error")}
        url="https://cdnjs.cloudflare.com/ajax/libs/phaser/2.6.2/phaser.min.js"
      />
      <div id="words-maze" style={style} />
    </div>
  );

  return (
    <div className={classes.headspace}>
      <Typography variant="h5" gutterBottom>
        <SplitHilite str={title} />
      </Typography>
      <Typography variant="body2" gutterBottom>
        {description}
      </Typography>
      <Typography variant="body2" gutterBottom>
        {instructions}
      </Typography>
      <MemberGate content={display} userAuth={props.userAuth} />
    </div>
  );
}

export default withTheme(Maze);
