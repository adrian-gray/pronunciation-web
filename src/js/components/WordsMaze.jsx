import React from 'react'
import Script from 'react-load-script'

import WordsMaze from './WordsMaze/index'

const NATIVE_WIDTH = 750
const NATIVE_HEIGHT = 750

class Maze extends React.Component {
  constructor (props) {
    super(props)

    this.state = {
      words: props.words
    }

    this.componentWillUnmount = this.componentWillUnmount.bind(this)
    this.handleScriptError = this.handleScriptError.bind(this)
    this.handleScriptLoad = this.handleScriptLoad.bind(this)
    this.render = this.render.bind(this)

    this._handleResize = this.handleResize.bind(this)
  }

  componentDidMount () {
    this.handleResize()
    this.windowListener = window.addEventListener('resize', this._handleResize)
  }

  componentWillUnmount () {
    window.removeEventListener('resize', this._handleResize)

    if (this && this.state && this.state.game) {
      this.state.game.destroy()
    } else {
      console.log('Unsuccessfully tried to destroy Words Maze', this.state)
    }
  }

  handleScriptError () {
    this.forceUpdate()
  }

  handleScriptLoad () {
    const wordsMaze = Object.create(WordsMaze)
    this.setState({
      game: wordsMaze
    })

    if (this && this.state) {
      wordsMaze.init({
        words: this.state.words
      })
    } else {
      console.log('KWords Maze state no longer exists, move on without creating game')
      // debugger
    }
  }

  handleResize () {
    const padding = '30px'
    const w = window.innerWidth - padding
    const scale = w / NATIVE_WIDTH
    const width = Math.round(scale * NATIVE_WIDTH)
    const height = Math.round(scale * NATIVE_HEIGHT)
    this.setState({ width, height })
  }

  render () {
    const width = `${this.state.width}px`
    const height = `${this.state.height}px`
    const style = { width, height }

    return (
      <div className='words-maze'>
        <Script
          onLoad={this.handleScriptLoad}
          onError={this.handleScriptError}
          url='https://cdnjs.cloudflare.com/ajax/libs/phaser/2.6.2/phaser.min.js'
        />
        <div id='words-maze' style={style} />
      </div>
    )
  }
}

export default Maze
