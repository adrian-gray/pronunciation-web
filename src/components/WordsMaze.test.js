import React from "react";
import { render, cleanup } from "@testing-library/react";

import WordsMaze from "./WordsMaze";

afterEach(cleanup);

test("WordsMaze", async () => {
  const props = {
    isUserAuth: true,
    tag: "bob",
    ipa: "b~",
    words: ["cat", "dog", "mouse", "bunny", "shoe"],
    correct: ["dog"]
  };

  const { container } = render(<WordsMaze {...props} />);

  expect(container).toMatchSnapshot();
});
