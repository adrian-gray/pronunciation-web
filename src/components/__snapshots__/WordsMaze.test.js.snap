// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WordsMaze 1`] = `
<div>
  <div
    class="head-space"
  >
    <h3>
      <span>
        Words Maze bob sound – b
      </span>
      <span
        class="hilite"
      />
       
    </h3>
    <p>
      Discover your way through the maze. Find the words that share the bob sound and travel along the path of the maze from one word to the next until you discover the way out. Master each level and advance to the next – more complex words, more twists and turns. Maze is an interactive pronunciation game that helps you become familiar with common English words so you can speak with confidence.
    </p>
    <p>
      From the yellow square pick an adjacent square with a b~ sound. Keep going until you reach the blue square.
    </p>
    <div>
      <div
        id="words-maze"
        style="width: 750px; height: 750px;"
      />
    </div>
  </div>
</div>
`;
